From 38bb76948e03461dd824eb5e8969d08b7a650f60 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Mon, 31 Mar 2025 16:16:23 +0100
Subject: [PATCH] dtoverlays: imx415: Add an override for 37.125MHz clock

Changing clock source also requires changing the link frequency.
Add a single override that does both for Waveshare modules
that are now using 37.125MHz clocks instead of 24MHz.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/README             | 3 +++
 arch/arm/boot/dts/overlays/imx415-overlay.dts | 2 ++
 2 files changed, 5 insertions(+)

--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -2873,6 +2873,9 @@ Params: addr                    Set I2C
         cam0                    Adopt the default configuration for CAM0 on a
                                 Compute Module (CSI0, i2c_vc, and cam0_reg).
         vcm                     Enable ad5398 VCM associated with the sensor.
+        clk-37125               Alternative to clock-frequency=37125000 and
+                                link-frequency=445500000 for Waveshare modules
+                                with 37.125MHz clock sources.
 
 
 Name:   imx462
--- a/arch/arm/boot/dts/overlays/imx415-overlay.dts
+++ b/arch/arm/boot/dts/overlays/imx415-overlay.dts
@@ -103,6 +103,8 @@
 		      <&cam_node>,"lens-focus:0=", <&vcm>;
 		clock-frequency = <&cam_clk>, "clock-frequency:0";
 		link-frequency = <&cam_endpoint>,"link-frequencies#0";
+		clk-37125 = <&cam_endpoint>,"link-frequencies#0=445500000",
+			   <&cam_clk>, "clock-frequency:0=37125000";
 		4lane = <0>, "+201+202";
 	};
 };
